src/Functions/FunctionsMergeTreePartInfo.cpp

```diff
@@ -0,0 +1,170 @@
+
+    bool useDefaultImplementationForLowCardinalityColumns() const override { return false; }
+    bool useDefaultImplementationForSparseColumns() const override { return false; }
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo &) const override { return false; }
+    size_t getNumberOfArguments() const override { return 1; }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        validateFunctionArguments(*this, arguments, FunctionArgumentDescriptors{
+            {"part_name", static_cast<FunctionArgumentDescriptor::TypeValidator>(&isAnyStringType), nullptr, "String or FixedString or LowCardinality String"}
+        });
+
+        return std::make_shared<ReturnType>();
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        auto result_column = ReturnColumn::create();
+        const IColumn * input_column = arguments[0].column.get();
+
+        Extractor extractor;
+        for (size_t i = 0; i < input_rows_count; ++i)
+        {
+            const MergeTreePartInfo part_info = constructPartInfo(input_column->getDataAt(i).toView());
+            extractor(part_info, *result_column);
+        }
+
+        return result_column;
+    }
+};
+
+struct NamePartitionId { static constexpr auto name{"partPartitionId"}; };
+struct NameMinBlock { static constexpr auto name{"partMinBlock"}; };
+struct NameMaxBlock { static constexpr auto name{"partMaxBlock"}; };
+struct NameMergeLevel { static constexpr auto name{"partMergeLevel"}; };
+struct NameMutationLevel { static constexpr auto name{"partMutationLevel"}; };
+
+}
+
+REGISTER_FUNCTION(MergeTreePartInfoTools)
+{
+    constexpr static auto partition_id_extractor = [](const MergeTreePartInfo & part_info, ColumnString & result_column)
+    {
+        result_column.insertData(part_info.getPartitionId().data(), part_info.getPartitionId().size());
+    };
+
+    constexpr static auto min_block_extractor = [](const MergeTreePartInfo & part_info, ColumnInt64 & result_column)
+    {
+        result_column.insertValue(part_info.min_block);
+    };
+
+    constexpr static auto max_block_extractor = [](const MergeTreePartInfo & part_info, ColumnInt64 & result_column)
+    {
+        result_column.insertValue(part_info.max_block);
+    };
+
+    constexpr static auto merge_level_extractor = [](const MergeTreePartInfo & part_info, ColumnInt64 & result_column)
+    {
+        result_column.insertValue(part_info.level);
+    };
+
+    constexpr static auto mutation_extractor = [](const MergeTreePartInfo & part_info, ColumnInt64 & result_column)
+    {
+        result_column.insertValue(part_info.mutation);
+    };
+
+    factory.registerFunction<FunctionMergeTreePartInfoExtractor<NamePartitionId, DataTypeString, ColumnString, decltype(partition_id_extractor)>>();
```

Comment:
I will remember that way how to pack lambda to the type inside template. Really nice.

https://fekir.info/post/decltype-lambda/
You could even call it from labda itself, there is no need to create an instance. And no need to make them `constexpr static`.